tensorboard/backend/http_util.py

```diff
@@ -157,6 +163,22 @@ def Respond(request,
   else:
     headers.append(('Expires', '0'))
     headers.append(('Cache-Control', 'no-cache, must-revalidate'))
+  if mimetype == _HTML_MIMETYPE and csp_scripts_sha256s:
+    # The quotes need to be single quotes per csp requirement.
+    script_srcs = ' '.join(
+        ["'sha256-{}'".format(sha256) for sha256 in csp_scripts_sha256s])
+    csp_string = ';'.join([
+            "default-src 'self'",
+            # data uri used by favicon
+            "img-src 'self' data:",
+            # gstatic: used by google-chart
+            # inline styles: Polymer templates + d3 uses inline styles.
+            "style-src https://www.gstatic.com data: 'unsafe-inline'",
+            # TODO(stephanwlee): remove `'strict dynamic'` when dynamic plugin
+            # resources can be hashed upfront.
+            "script-src 'strict-dynamic' %s" % script_srcs,
+        ])
+    headers.append(('Content-Security-Policy', csp_string))
```

Comment:
FWIW, I ran this through https://csp-evaluator.withgoogle.com/ and it reports a few other things (also mentioned internally):

- Missing base-uri allows the injection of base tags. They can be used to set the base URL for all relative (script) URLs to an attacker controlled domain. Can you set it to 'none' or 'self'?
- Can you restrict object-src to 'none'?

Reply:
We do make use of the `base` so I changed it to an explicit `self` (it used to be implicit `self` by `default-src`). Done!